### ISSUE TITLE
Remaining issues with docker development

### ISSUE THREAD

**Issue by exbinary** [rubyforgood/mutual-aid#718]
Documenting a few outstanding issues with dockerized development:

1. Running `bin/dev/test` generates the following warning:
```
ActiveRecord::EnvironmentMismatchError: You are attempting to modify a database that was last run in `development` environment.
You are running in `test` environment. If you are sure you want to continue, first set the environment using:
        rails db:environment:set RAILS_ENV=test
```
Steps to recreate: Run `bin/dev/bootstrap`, then `bin/dev/test`.

1. Could be worth silencing warnings for now. There are still a few outstanding gems (eg `mobility`) that haven't fixed their warnings generated by newer rubies.

**Comment by indiebrain** [comment]
A few notes on what I _think_ I've found so far...

> ActiveRecord::EnvironmentMismatchError:

TLDR; I suspect that the `development` and `test` environments are sharing a database. 

It seems that the development `docker-compose.yml` sets a [default `POSTGRES_DB` ENV var value of `mutualaid`](https://github.com/rubyforgood/mutual-aid/blob/main/docker/development/docker-compose.yml#L3). This passes through to the application, and takes precedence in both the [`development`](https://github.com/rubyforgood/mutual-aid/blob/main/config/database.yml#L40) AND [`test`](https://github.com/rubyforgood/mutual-aid/blob/main/config/database.yml#L51) - IE development and test both use the `mutualaid` database. I currently believe this to be the cause of the `ActiveRecord::EnvironmentMismatchError`, and will put together a change set to address this.

**Comment by grenewode** [comment]
@indiebrain - I think your interpretation is correct. When I wrote these, I think I was only testing it against the [`docker-development`](https://github.com/rubyforgood/mutual-aid/blob/14bb7269c30d712377d4f9d4337090203fc03f0e/config/database.yml#L42-L44) configuration. When I originally the `docker-development` configuration, the  the other configurations didn't pull values from environment variables:
https://github.com/rubyforgood/mutual-aid/blob/e10d0b05fd122130279d849b60e60ea80308caa8/config/database.yml#L24-L977
Since I didn't (and still don't) know that much about ruby configuration files, I didn't want to change the other configurations, so I created a new one for docker instead. At this point, I think you might be able to get rid of the `docker-environment` configuration altogether. It is only referenced in three places in the project. One of them is the database.yml configuration for docker your already looking at. I think others could be eliminated/unified as well:
1. https://github.com/rubyforgood/mutual-aid/blob/0a84965ce7c1da34afb1ef24aa62b74eb63dc74f/ansible/files/database.yml#L33-L50
Basically the same (maybe should be identical/symlinked to) the docker configuration, but less up to date. I think this was mostly done by @padthaitofuhot when we were setting up LAMA's instances on ansible.
2. https://github.com/rubyforgood/mutual-aid/blob/0a84965ce7c1da34afb1ef24aa62b74eb63dc74f/config/webpacker.yml#L52-L89 
More complicated because of the webpack development server. It looks like the `docker-development` configuration is identical to the `development` configuration, except that the docker config has the `webpack-dev-server` disabled. I think I disabled the dev server because it doesn't really serve any purpose in a immutable docker image. However, if people are already using the vanilla development configuration with docker, and the dev server isn't causing any issues, there's no reason to bother turning it off.

**Comment by indiebrain** [comment]
@grenewode, thanks for the context - I'm still supa-green here, and ANY feedback is most appreciated. 

I think the `development-docker` config bits can go, they seem to be vestigial configuration and don't seem to be used. I'll put together a PR to remove them. I agree that the Ansible config might as well be a symlink to the native config, but I'm not big-city Ansible expert so I'll probably rely on the principal of "All of us are better than ANY of us" during code review.
